addons/im_livechat/models/discuss_channel.py

```diff
@@ -254,6 +254,21 @@ def _gc_empty_livechat_sessions(self):
         empty_channel_ids = [item['id'] for item in self.env.cr.dictfetchall()]
         self.browse(empty_channel_ids).unlink()
 
+    @api.autovacuum
+    def _gc_bot_only_ongoing_sessions(self):
+        """ Garbage collect bot-only livechat sessions with no activity for over 1 day."""
+        limit_date = fields.Datetime.subtract(fields.Datetime.now(), days=1)
+
+        stale_sessions = self.search([
+            ("channel_type", "=", "livechat"),
+            ("livechat_active", "=", True),
+            ("last_interest_dt", "<=", limit_date),
+            ("livechat_agent_partner_ids", "=", False),
+        ]).filtered("livechat_bot_history_ids")
```

Comment:
I don't like the idea of fetching channels, only to filter them later on. Especially since there might be a *lot* of them :) What case are we actually covering with this filter? I would expect the session to be "chat bot only" if no agent was ever assigned?